### ISSUE TITLE
Issue: Compensable activity not working

### ISSUE THREAD

**Issue by VartuSharma** [elsa-workflows/elsa-core#4342]
Hello @sfmskywalker 
Using Compensable activity, I developed a single straightforward procedure (see json file attached). It functioned properly in earlier versions. However, it does not function properly with **Elsa 2.12**, the most recent version.
Added fault state in body section, still compensable part is not executed.

Kindly assist! As I need this activity in my current task.
[compensable-example.zip](https://github.com/elsa-workflows/elsa-core/files/12393100/compensable-example.zip)
![image](https://github.com/elsa-workflows/elsa-core/assets/127403724/4754f732-4672-46c0-806b-024ec59dc26f)

**Comment by VartuSharma** [comment]
Hello Any updates on this bug?

**Comment by mohdali** [comment]
@VartuSharma I cannot see any changes done between 2.11 to 2.12 that could have affected Compensable activities. I've also tried your example and found it is working fine (in my understanding). Please check below sample executions.

Compensate after activity is done:
![image](https://github.com/elsa-workflows/elsa-core/assets/4145465/2c56f4f9-218c-44fe-b045-1d061b14a763)

Cancel due to Fault:
![image](https://github.com/elsa-workflows/elsa-core/assets/4145465/c4cda842-e9b9-40cf-bd25-76d1e9b29453)

Please note that putting Compensate inside Body will cause Cancel to be called instead. I believe this is by design.

**Comment by VartuSharma** [comment]
Hello @mohdali 
Let me try again.

Thanks

**Comment by mohdali** [comment]
> Hello @mohdali Let me try again. In which version of Elsa you tried?
> 
> Thanks

I tested with latest code from master. It is also now released as 2.13.0, please give it a try.

However, I don't see any recent changes that could have impacted Compensable activities so I'm curious what could be the issue.

**Comment by hsnsalhi** [comment]
hello, having same issue with 2.12.0.
I tried the same example as @VartuSharma

**Comment by VartuSharma** [comment]
@mohdali @hsnsalhi @sfmskywalker 

Compensation is working with 2.13.0, the latest version.  But not with 2.12.0/2.11.

**Comment by sfmskywalker** [comment]
Closing because this was confirmed fixed in Elsa 2.13.0. The thread shows maintainer retesting on newer code and the reporter later confirmed: "Compensation is working with 2.13.0".
